[DOCS] Adds a consolidated overview of using GX as a conceptual guide

Rachel-Reverie opened the pull request:
## Description
- Consolidates information to provide a high level overview of GX's major components and how they function in a basic GX workflow.

## Definition of done
- [X] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [ ] Appropriate tests and docs have been updated

Rachel-Reverie committed:
First draft, GX overview

ghost commented:
<details open>
<summary><b>👇 <a href="https://app.codesee.io/r/reviews?pr=8045&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations">Click on the image</a> for a new way to code review</b>
</summary>

####

[![Review these changes using an interactive CodeSee Map](https://s3.us-east-2.amazonaws.com/maps.codesee.io/images/github/great-expectations/great_expectations/8045/e2789072/fc372bf6e29112de2aa8dab50e8af60f4f306689.svg)](https://app.codesee.io/r/reviews?pr=8045&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations)

#### Legend
<img src="https://app.codesee.io/Legend_v3.png" alt="CodeSee Map legend" width="220" height="156">
</details>

Rachel-Reverie committed:
Second draft edits
- updates the python version in _data.jsx
- imports python version for the guide
- links guide to the Expectations Gallery

Rachel-Reverie committed:
Second draft edits
- updates the python version in _data.jsx
- imports python version for the guide
- links guide to the Expectations Gallery

Rachel-Reverie committed:
Adds overview to the ToC

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,118 @@
+
+A **Checkpoint** is the primary means for validating data in a production deployment of GX.  Checkpoints provide an abstraction for bundling a Batch (or Batches) of data with an Expectation Suite (or several), and then running those Expectations against the paired data.  
+
+One of the most powerful features of Checkpoints is that they can be configured to run **Actions**, which will do some process based on the Validation Results generated when a Checkpoint is run. Typical uses include sending email, slack, or custom notifications. Another common use case is updating Data Docs sites. However, Actions can be created to do anything you are capable of programing in Pyt ...
```
```suggestion
One of the most powerful features of Checkpoints is that you can configure them to run Actions. The Validation Results generated when a Checkpoint runs determine what Actions are performed. Typical use cases include sending email, Slack, or custom notifications. Another common use case is updating Data Docs sites. Actions can be used to do anything you are capable of programming in Python. Actions are a versatile tool for integrating Checkpoints in your pipeline's workflow.
```

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,118 @@
+---
+title: GX Overview
```
```suggestion
title: Great Expectations Overview
```
It would be great if we could move away from these types of titles (Overview). Perhaps _Great Expectations fundamentals_,  _Great Expectations basics_,  _An introduction to Great Expectations_, _Get to know Great Expectations_? The title should be more informative and inviting. 

We should use _Great Expectations_ on first use and not _GX_. This is first use. Also, I recommend defining the acronym in the body text on first use.  For example: Great Expectations (GX). GX can then be used after the acronym has been defined.

Rachel-Reverie replied:
Good points, all.  The updates will be included in my next commit, rather than excepting just the suggestion here.

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,118 @@
+
+import PythonVersion from '/docs/components/_python_version.mdx';
+
+Great Expectations is a framework for defining Expectations and running them against your data. Like assertions in traditional Python unit tests, Expectations provide a flexible, declarative language for describing expected behavior. Unlike traditional unit tests, GX applies Expectations to data instead of code. For example, you could define an Expectation that a column contains no null values, a ...
```
```suggestion
Great Expectations is a framework for defining and running Expectations on your data. Similar to assertions in Python unit tests, Expectations provide a flexible, declarative language for describing the expected behavior. Unlike traditional unit tests, Expectations are applied to data and not code. For example, you can define an Expectation that a column contains no null values, the Expectation runs on your data, and a report is generated when a null value is found.
```

"..describing the expected behavior" of what?

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,118 @@
+
+Great Expectations is a framework for defining Expectations and running them against your data. Like assertions in traditional Python unit tests, Expectations provide a flexible, declarative language for describing expected behavior. Unlike traditional unit tests, GX applies Expectations to data instead of code. For example, you could define an Expectation that a column contains no null values, a ...
+
+## Before you start
```
```suggestion
## Source data
```

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,118 @@
+
+## Before you start
+
+Before you've even installed GX, working with GX starts with data.  Specifically, data you care about enough for quality to be important.  We call your data, in its original format, your **Source Data**.  The combination of your data in its original format and the location at which it is stored is your **Source Data System**.
```
```suggestion
GX uses the term **Source Data** when referring to data in its original format, and the term **Source Data System** when referring to the storage location for Source Data. No matter the format of your Source Data System, or where it's hosted, you can use GX to ensure the quality and consistency of your Source Data.

```
We need to stop mixing first-person and first-person plural pronouns and get away from using _we_. When using _we_, the antecedent for the pronoun must be clear. I recommend avoiding _we_.

HaebichanGX replied:
Hey @Rachel-Reverie I think this is great initial work. It gets specific into a lot of the what to do's of GX and what different parts are. But as a first time user of GX, I feel like I'd be equally if not more curious about "Why GX"? Take a step back and clarifying the purpose of using GX - specific benefits and unique advantages of using GX of over frameworks - in the initial parts of the text would really hook the readers. 

I sort of lost focus after paragraph 5 since there was too much text my attention span isn't great with writing and is usually lengthened through visual guides.

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,118 @@
+
+Before you've even installed GX, working with GX starts with data.  Specifically, data you care about enough for quality to be important.  We call your data, in its original format, your **Source Data**.  The combination of your data in its original format and the location at which it is stored is your **Source Data System**.
+
+Your Source Data System could be CSV files in a folder, or a PostgreSQL database hosted on AWS, or any combination of numerous formats and environments.  Regardless of what your Source Data System is or where it resides, your next step in assuring its quality is implementing GX.
```
```suggestion

```

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,118 @@
+
+Your Source Data System could be CSV files in a folder, or a PostgreSQL database hosted on AWS, or any combination of numerous formats and environments.  Regardless of what your Source Data System is or where it resides, your next step in assuring its quality is implementing GX.
+
+## Getting started with Great Expectations
```
```suggestion
## Get started with Great Expectations
```
I recommend that we avoid using gerunds in headings.

![image](https://github.com/great-expectations/great_expectations/assets/104205257/03ae93ae-c790-42e9-894e-73f76d04dfc0)

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,118 @@
+
+## Getting started with Great Expectations
+
+You have data.  You care about your data.  You've made the wise decision to use Great Expectations to catch potential quality concerns with your data.  So the next thing you need to know is: _What next?_
```
```suggestion

```

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,118 @@
+
+You have data.  You care about your data.  You've made the wise decision to use Great Expectations to catch potential quality concerns with your data.  So the next thing you need to know is: _What next?_
+
+Getting started with GX can be broken down into four steps.  First, you will install GX and create a GX project.  Next, you will connect that project to your Source Data System.  Third, you will define your Expectations for the connected data. Finally, you will use GX to Validate those Expectations.
```
```suggestion
To get started with GX, you complete the following tasks: 

- Install GX and create a GX project.  

- Connect the GX project to your Source Data System.  

- Define your Expectations for the Source Data. 

- Validate the Expectations.

```

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,118 @@
+
+Getting started with GX can be broken down into four steps.  First, you will install GX and create a GX project.  Next, you will connect that project to your Source Data System.  Third, you will define your Expectations for the connected data. Finally, you will use GX to Validate those Expectations.
+
+Installation and Setup -> Connect to data -> Create Expectations -> Validate data
```
```suggestion

```
I assume this is a placeholder is for a workflow image. I'm not sure an image is necessary with a simple, straightforward bulleted task list.

Rachel-Reverie replied:
It's more to serve as a callback for a later image, where this "first time process" connects with the iterative workflow.

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,118 @@
+
+Installation and Setup -> Connect to data -> Create Expectations -> Validate data
+
+### Install Great Expectations
```
```suggestion
### Install GX
```

HaebichanGX replied:
I'm not sure why we need information on installing great expectations here? This feels more like for a how to guide like a quickstart for specific implementation, which we already have on our Quickstart.

HaebichanGX commented:
Hey @Rachel-Reverie I think this is great initial work. It gets specific into a lot of the what to do's of GX and what different parts are. But as a first time user of GX, I feel like I'd be equally if not more curious about "Why GX"? Take a step back and clarifying the purpose of using GX - specific benefits and unique advantages of using GX of over frameworks - in the initial parts of the text would really hook the readers.

I sort of lost focus after half way in since there was too much text my attention span isn't great with writing and is usually lengthened through visual guides.

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,118 @@
+
+### Install Great Expectations
+
+GX is a library for <PythonVersion />, and can be installed using `pip`, a library installer bundled with Python installations.  Depending on your Source Data System, you may need to install GX with some additional dependencies (generally for SQL Source Data Systems, or environments like AWS).  We streamline this process by providing special `pip` commands for specific combinations of additional  ...
```
```suggestion
GX is a library for <PythonVersion />, and you can use pip to install it. If you're using a Source Data System such as SQL or AWS, special  pip commands are provided for the installation of additional dependencies.
```
What are _additional dependencies_? Can this be defined or clarified?

Note the convention for _pip_ on the pip website:

![image](https://github.com/great-expectations/great_expectations/assets/104205257/502093df-6d8a-4f21-a320-83e8e9df54f2)

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,118 @@
+
+GX is a library for <PythonVersion />, and can be installed using `pip`, a library installer bundled with Python installations.  Depending on your Source Data System, you may need to install GX with some additional dependencies (generally for SQL Source Data Systems, or environments like AWS).  We streamline this process by providing special `pip` commands for specific combinations of additional  ...
+
+For details on how to install GX and how to include any additional dependencies for your Source Data System, see [our installation guides](landing page: Setting up a GX environment->Installation and Dependencies).
```
```suggestion
To install GX and additional dependencies, see [Installation guides](landing page: Setting up a GX environment->Installation and Dependencies).
```

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,118 @@
+For details on how to install GX and how to include any additional dependencies for your Source Data System, see [our installation guides](landing page: Setting up a GX environment->Installation and Dependencies).
+
+
+### Set up your GX Data Context
```
```suggestion
### Configure your Data Context
```

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,118 @@
+
+### Set up your GX Data Context
+
+A Great Expectations project stores its configurations in a **Data Context**.  In Python, the Data Context object also serves as your entry point for the GX API, allowing you to further configure and interact with GX.  Your Data Context manages various classes and helps limit the objects you need to directly manage to get GX working on your data.
```
```suggestion
A GX project configuration is stored in a **Data Context**. In Python, the Data Context object also serves as your entry point for the GX API, allowing you to further configure and interact with GX.  Your Data Context manages various classes and limits the objects you need to manage to get GX working.
```

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,118 @@
+
+A Great Expectations project stores its configurations in a **Data Context**.  In Python, the Data Context object also serves as your entry point for the GX API, allowing you to further configure and interact with GX.  Your Data Context manages various classes and helps limit the objects you need to directly manage to get GX working on your data.
+
+GX provides three types of Data Contexts:
```
```suggestion
The following are the available Data Contexts types:
```

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,118 @@
+A Great Expectations project stores its configurations in a **Data Context**.  In Python, the Data Context object also serves as your entry point for the GX API, allowing you to further configure and interact with GX.  Your Data Context manages various classes and helps limit the objects you need to directly manage to get GX working on your data.
+
+GX provides three types of Data Contexts:
+- An **Ephemeral Data Context**: This Data Context exists in memory, and will not persist beyond the current Python session.
```
```suggestion
- **Ephemeral Data Context**: Exists in memory, and does not persist beyond the current Python session.

```

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,118 @@
+
+GX provides three types of Data Contexts:
+- An **Ephemeral Data Context**: This Data Context exists in memory, and will not persist beyond the current Python session.
+- A **Filesystem Data Context**: This Data Context exists as a folder and configuration files, allowing its contents to persist between Python sessions.
```
```suggestion
- **Filesystem Data Context**: Exists as a folder and configuration files, and its contents persist between Python sessions.
```

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,118 @@
+GX provides three types of Data Contexts:
+- An **Ephemeral Data Context**: This Data Context exists in memory, and will not persist beyond the current Python session.
+- A **Filesystem Data Context**: This Data Context exists as a folder and configuration files, allowing its contents to persist between Python sessions.
+- A **Cloud Data Context**: This Data Context also supports persistence between Python sessions, but additionally serves as the entry point for GX Cloud.
```
```suggestion
- **Cloud Data Context**: Supports persistence between Python sessions, but additionally serves as the entry point for Great Expectations Cloud.
```
Great Expectations Cloud is a product name (?) and shouldn't be truncated.

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,118 @@
+- A **Filesystem Data Context**: This Data Context exists as a folder and configuration files, allowing its contents to persist between Python sessions.
+- A **Cloud Data Context**: This Data Context also supports persistence between Python sessions, but additionally serves as the entry point for GX Cloud.
+
+<!-- For details on how to initialize and instantiate a Data Context, see [our Data Context guides](landing page: Setting up a GX environment->Data Contexts). -->
```
```suggestion
<!-- To initialize and instantiate a Data Context, see [Setting up a GX environment](Setting up a GX environment). -->
```

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,118 @@
+<!-- For details on how to initialize and instantiate a Data Context, see [our Data Context guides](landing page: Setting up a GX environment->Data Contexts). -->
+
+
+#### Optional configurations for metadata Stores and Data Docs
```
Stores and Data Docs haven't been defined or introduced previously. Why are they important? How are they used? Where are they used? Are they optional?

These terms need to be introduced and discussed before you instruct users how to perform optional configurations.

Rachel-Reverie replied:
I'm adding a brief explanation of these configurations in the section on Data Contexts.

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,118 @@
+
+#### Optional configurations for metadata Stores and Data Docs
+
+**Stores** are connectors to the metadata GX uses.  This includes things like:
```
```suggestion
**Stores** are metadata connectors that can provide the following information:
```

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,118 @@
+**Stores** are connectors to the metadata GX uses.  This includes things like:
+- The Datasource and Data Asset configurations that tell GX how to connect to your Source Data System.
+- The Expectations you have specified about your Source Data.
+- The Metrics that are recorded when GX validates certain Expectations against data.
```
```suggestion
- The Metrics that are recorded when GX validates certain Expectations against your Source Data.
```

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,118 @@
+- The Checkpoints you have configured for validating your Expectations.
+- The Validation Results that GX returns when you run a Checkpoint.
+
+A GX Data Context comes with default configurations in place for your Stores.  By default, these exist in the same folder hierarchy as your Data Context (or in memory, if you are using an Ephemeral Data Context).  However, you can update these if you want to change where your Store configurations exist, such as hosting them in a shared cloud environment like AWS.
```
```suggestion
A Data Context includes the default configurations for your Stores. Default configurations are stored with your Data Context, or in memory if you're using an Ephemeral Data Context. You can move your configurations to another storage location, or you can host them in a shared cloud environment such as AWS.
```
Very confused:

A **Data Context** includes default configurations for your Stores.  By default, **these exist in the same folder hierarchy as your Data Context** (or in memory, if you are using an Ephemeral Data Context). 

So, the Data Context includes default configurations and they are are stored with the Data Context? This needs to be clarified. I've made an attempt.

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,118 @@
+
+A GX Data Context comes with default configurations in place for your Stores.  By default, these exist in the same folder hierarchy as your Data Context (or in memory, if you are using an Ephemeral Data Context).  However, you can update these if you want to change where your Store configurations exist, such as hosting them in a shared cloud environment like AWS.
+
+<!-- For more information on configuring Stores, see [our guides on metadata Stores](landing page: setting up a GX environment->Metadata Stores). -->
```
```suggestion
<!--To configure Stores, see [Metadata Stores](landing page: setting up a GX environment->Metadata Stores). -->
```

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,118 @@
+
+<!-- For more information on configuring Stores, see [our guides on metadata Stores](landing page: setting up a GX environment->Metadata Stores). -->
+
+**Data Docs** are human-readable documentation about your Expectations and Validation Results that GX generates for you.  They exist as webpages, and like Stores your Data Context will include a default configuration for building them locally for Filesystem Data Contexts, in memory for Ephemeral Data Contexts, and online for Cloud Data Contexts.  As with Stores, you can configure where your Data  ...
```
```suggestion
**Data Docs** are human-readable documentation about your Expectations and Validation Results.  They exist as webpages, and your Data Context includes a default configuration for building them locally for Filesystem Data Contexts, in memory for Ephemeral Data Contexts, and online for Cloud Data Contexts. You can configure where your Data Docs are hosted.  You can also add configurations for multiple Data Docs sites, and specify what information each Data Docs site provides.
```

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,118 @@
+
+**Data Docs** are human-readable documentation about your Expectations and Validation Results that GX generates for you.  They exist as webpages, and like Stores your Data Context will include a default configuration for building them locally for Filesystem Data Contexts, in memory for Ephemeral Data Contexts, and online for Cloud Data Contexts.  As with Stores, you can configure where your Data  ...
+
+<!-- For more information on configuring Data Docs, see our [Data Docs guides](landing page:Setting up a GX environment->Data Docs). -->
```
```suggestion
<!--To configure Data Docs, see [Data Docs](landing page:Setting up a GX environment->Data Docs). -->
```

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,118 @@
+
+### Connect GX to your Source Data System
+
+**Datasources** provide the interface between GX and your Source Data System.  Regardless of the original format of your Source Data, a Datasource will provide you a clear and consistent API for accessing that data in GX.
```
```suggestion
**Datasources**connect GX to your Source Data System.  Regardless of the original format of your Source Data, a Datasource let's you access it.
```

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,118 @@
+
+**Datasources** provide the interface between GX and your Source Data System.  Regardless of the original format of your Source Data, a Datasource will provide you a clear and consistent API for accessing that data in GX.
+
+**Data Assets** are collections of records within a Datasource.  These collections can be further partitioned to a desired specification.  We call these subsets of data **Batches**.  For instance, you could define a Data Asset for a SQL Datasource as the selection of all records from last year in a given table.  Then you could further partition that Data Asset into Batches of data corresponding t ...
```
```suggestion
**Data Assets** are collections of records within a Datasource.  You can further partition these collections of records into subsets of data that GX calls **Batches**.  For instance, you could define a Data Asset for a SQL Datasource as the selection of all records from last year in a given table. You could then partition that Data Asset into Batches of data that correspond to the records for individual months of the year.
```

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,118 @@
+
+**Data Assets** are collections of records within a Datasource.  These collections can be further partitioned to a desired specification.  We call these subsets of data **Batches**.  For instance, you could define a Data Asset for a SQL Datasource as the selection of all records from last year in a given table.  Then you could further partition that Data Asset into Batches of data corresponding t ...
+
+How you organize your Datasources into Data Assets and your Data Assets into Batches will impact how you can analyze them with GX.  Defining multiple Batches in a Data Asset will let GX automate the process of running statistical analyzes for each Batch.  It will also let you specify an individual Batch to work with.  In the example of a Data Asset that has been split into months, this would let  ...
```
```suggestion
How you organize your Datasources into Data Assets and your Data Assets into Batches determines how you can analyze them with GX.  Defining multiple Batches in a Data Asset lets GX automate the process of running statistical analyzes for each Batch.  It also lets you specify an individual Batch to work with.  In the example of a Data Asset that has been split into months, this would let you build a statistical model for each month of your existing data.  Then, after an update, you could specify that you only wanted to run your analysis on the most recent month's data.
```

This is heavy going. Is there any way we can clarify and lighten it?

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,118 @@
+
+How you organize your Datasources into Data Assets and your Data Assets into Batches will impact how you can analyze them with GX.  Defining multiple Batches in a Data Asset will let GX automate the process of running statistical analyzes for each Batch.  It will also let you specify an individual Batch to work with.  In the example of a Data Asset that has been split into months, this would let  ...
+
+<!-- For more information on Datasources and Data Assets, see our [Connecting to data guides](Landing page: connecting to data). -->
```
```suggestion
<!-- For more information about Datasources and Data Assets, see [Connecting to data](Landing page: connecting to data). -->
```

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,118 @@
+
+### Define your Expectations
+
+An **Expectation** is a verifiable assertion about data.  Expectations enhance communication about your data by describing the state it should conform to.  They also improve quality for data applications by giving you a verifiable standard to measure your data against.  They help you take the implicit assumptions about your data and make them explicit.  Having defined Expectations helps reduce th ...
```
```suggestion
An **Expectation** is a verifiable assertion about Source Data.  Expectations enhance communication about your data by describing the state it should conform to.  They also improve quality for data applications by giving you a verifiable standard to measure your data against.  They help you take the implicit assumptions about your data and make them explicit.  Having defined Expectations helps reduce the need to consult with domain experts about uncertainties and lets you avoid leaving insights about data in isolated silos.
```
"...enhance communication about your data" ????

"...avoid leaving insights about data in isolated silos" ????

Rachel-Reverie replied:
I tried to add some clarifications there.

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,118 @@
+
+An **Expectation** is a verifiable assertion about data.  Expectations enhance communication about your data by describing the state it should conform to.  They also improve quality for data applications by giving you a verifiable standard to measure your data against.  They help you take the implicit assumptions about your data and make them explicit.  Having defined Expectations helps reduce th ...
+
+Expectations can be built off of the domain knowledge of subject matter experts, interactively while introspecting a sample set of data, or by using one of GX's powerful tools: the Data Assistant.  A **Data Assistant** is a utility that automates the process of building Expectations by asking questions about your data, gathering information to describe what is observed, and then presenting Metric ...
```
```suggestion
 GX provides a Data Assistant to speed the creation of Expectations.  A Data Assistant is a utility that automates the process of building Expectations by asking questions about your data, gathering information to describe what is observed, and then presenting Metrics and proposed Expectations based on the answers it has determined.
```
Can we remove the bold from terms in this doc? Bold should be reserved for UI elements in tasks.

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,118 @@
+
+Expectations can be built off of the domain knowledge of subject matter experts, interactively while introspecting a sample set of data, or by using one of GX's powerful tools: the Data Assistant.  A **Data Assistant** is a utility that automates the process of building Expectations by asking questions about your data, gathering information to describe what is observed, and then presenting Metric ...
+
+GX's built-in library includes more than 50 common Expectations, such as:
```
```suggestion
The GX built-in library includes more than 50 Expectations, including:
```
Avoid the possessive form of the company name.

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,118 @@
+- `expect_table_row_count_to_be_between`
+- `expect_column_median_to_be_between`
+
+GX also allows you to create custom Expectations, should you need something more specialized than what is available.  You can even use custom Expectations contributed by other GX community members by installing those Expectations as Plugins.
```
```suggestion
You can create custom Expectations when you need something more specialized for your business requirements.  You can also use custom Expectations contributed by GX community members by installing them as Plugins.
```

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,118 @@
+
+GX also allows you to create custom Expectations, should you need something more specialized than what is available.  You can even use custom Expectations contributed by other GX community members by installing those Expectations as Plugins.
+
+For a full list of available Expectations, see [the Expectation Gallery](https://greatexpectations.io/expectations/).
```
```suggestion
For a list of available Expectations, see [the Expectation Gallery](https://greatexpectations.io/expectations/).
```

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,118 @@
+
+For a full list of available Expectations, see [the Expectation Gallery](https://greatexpectations.io/expectations/).
+
+<!-- For details on how to define Expectations, see our [Creating Expectations guides](landing page: Creating Expectations). -->
```
```suggestion
<!-- To define Expectations, see [Creating Expectations](landing page: Creating Expectations). -->
```

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,118 @@
+
+One of the most powerful features of Checkpoints is that they can be configured to run **Actions**, which will do some process based on the Validation Results generated when a Checkpoint is run. Typical uses include sending email, slack, or custom notifications. Another common use case is updating Data Docs sites. However, Actions can be created to do anything you are capable of programing in Pyt ...
+
+<!-- For more details on using Checkpoints to validate your data, see our [Validating data guides](landing page: validating data). -->
```
```suggestion
<!-- To use Checkpoints to validate your data, see [Validating data](landing page: validating data). -->
```

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,118 @@
+One of the most powerful features of Checkpoints is that they can be configured to run **Actions**, which will do some process based on the Validation Results generated when a Checkpoint is run. Typical uses include sending email, slack, or custom notifications. Another common use case is updating Data Docs sites. However, Actions can be created to do anything you are capable of programing in Pyt ...
+
+<!-- For more details on using Checkpoints to validate your data, see our [Validating data guides](landing page: validating data). -->
+<!-- For more details on integrating GX with your processes, see our [guide on including GX in a data pipeline](). -->
```
```suggestion
<!-- To integrate GX with your processes, see [guide on including GX in a data pipeline](). -->
```

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,118 @@
+
+#### Evaluate Validation Results
+
+The **Validation Results** returned by GX function as a report on how well your data actually corresponds to what you expected of it.  You will be able to view this information in the Data Docs that are configured in your Data Context.  Evaluating your Validation Results will help you spot issues with your data.  You can then use that knowledge to decide how to proceed.  If the Validation Results ...
```
```suggestion
The **Validation Results** returned by GX tell you how your data corresponds to what you expected of it.  You can view this information in the Data Docs that are configured in your Data Context.  Evaluating your Validation Results helps you identify issues with your data.  If the Validation Results show that your data meets your Expectations, you can confidently use it.
```

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,118 @@
+
+The **Validation Results** returned by GX function as a report on how well your data actually corresponds to what you expected of it.  You will be able to view this information in the Data Docs that are configured in your Data Context.  Evaluating your Validation Results will help you spot issues with your data.  You can then use that knowledge to decide how to proceed.  If the Validation Results ...
+
+If your data _fails_ to meet your Expectations, then you may determine that your data has a quality issue you need to address.  Or you may realize that your Expectations need to be refined to correspond to an unexpected reality reflected in your data.  After refining your data (or your Expectations) you can reuse the Checkpoint you had created to Validate your data once more.
```
```suggestion
If your data fails to meet your Expectations, your data might have a quality issue that you need to address, or you might need to modify your Expectations to correspond to an unexpected reality reflected in your data.  After refining your data or your Expectations you can reuse the Checkpoint you created to Validate your data.
```

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,118 @@
+
+## The iterative GX workflow
+
+Ensuring data quality is an ongoing process.  When new data comes in or existing data is updated, GX lets you re-use your existing Expectation Suites and Checkpoints to validate the new or updated Source Data.  This permits you to maintain records of your data's adherence to Expectations and track trends in the quality of the validated data.  These records can help you determine if failed Expecta ...
```
```suggestion
Ensuring data quality is an ongoing process.  When new data comes in or existing data is updated, GX lets you reuse your existing Expectation Suites and Checkpoints to validate the new or updated Source Data.  This lets you to maintain records of your data's adherence to Expectations and track trends in the quality of the validated data.  These records can help you determine if failed Expectations are due to outliers in new data, a more systemic problem, or the need to update your understanding of the data you are working with.
```

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,118 @@
+
+Ensuring data quality is an ongoing process.  When new data comes in or existing data is updated, GX lets you re-use your existing Expectation Suites and Checkpoints to validate the new or updated Source Data.  This permits you to maintain records of your data's adherence to Expectations and track trends in the quality of the validated data.  These records can help you determine if failed Expecta ...
+
+We call this cycle of validating data the iterative GX workflow.
```
```suggestion
This cycle of data validation is known as the iterative GX workflow.
```

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,118 @@
+
+ Validate Data -> Evaluate Validation Results (Accept data, improve data, or refine Expectations) -> Iterate (after you add new Datasources or update the data in existing ones)
+
+Whether Validation Results reveal that your data continues to meet defined expectations, that a problem exists with the quality of your data, or that your previous understanding of your data hasn't caught up with something that has changed: now you know!
```
```suggestion

```

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,118 @@
+
+Whether Validation Results reveal that your data continues to meet defined expectations, that a problem exists with the quality of your data, or that your previous understanding of your data hasn't caught up with something that has changed: now you know!
+
+You can use this knowledge to reject bad data, or to refine it and improve its quality to meet your standards.  Or you may use this knowledge to refine your Expectations to match your improved understanding of how your data has changed.  All of these uses are valuable parts of maintaining data quality over time.
```
```suggestion
You can use Validation Results to reject bad data, or to refine it and improve its quality to meet your standards. Alternatively, you can use Validation Results to refine your Expectations to match your improved understanding of how your data has changed. 
```

kwcanuck reviewed (approved):
Thanks for the effort @Rachel-Reverie!

Like @HaebichanGX, I think this is a good overview of GX and its components. However, I'm not sure how useful it will be to our users. Do developers or data scientists want to read a long narrative document about GX and its components? Would they prefer a brief use case analysis, a brief overview of the getting started process (with links), and examples of the problems that GX can solve? 

I agree with Haebichan that we would better serve our users if we clarify the purpose of GX and clearly articulate its benefits and advantages over other data validation methods. 

I'm not sure of the purpose of this doc, or who the audience might be. Just my two cents.

Rachel-Reverie committed:
Update docs/docusaurus/docs/conceptual_guides/gx_overview.md

Co-authored-by: Rob Gray <104205257+kwcanuck@users.noreply.github.com>

Rachel-Reverie committed:
updates per reviews

Rachel-Reverie committed:
update per reviews

Rachel-Reverie committed:
update per reviews

Rachel-Reverie committed:
add value prop and intended audience sections.

joshzzheng commented:
Some feedback:
1. Let's remove why GX section
2. Still too much wall of text
3. We need to have diagrams and pictures. At least one diagram showing the relationship between different components. Another diagram showing the major (4 step) workflow and how different key components fit into the 4 big steps (our user had created one that I shared)

Rachel-Reverie committed:
initial drafting for revised overview, multi-part

Rachel-Reverie committed:
minor edits

Rachel-Reverie committed:
Merge branch 'develop' into d/doc_483/consolidated_gx_overview_conceptual_guide

Rachel-Reverie committed:
Merge remote-tracking branch 'origin/d/doc_483/consolidated_gx_overview_conceptual_guide' into d/doc_483/consolidated_gx_overview_conceptual_guide

# Conflicts:
#	docs/docusaurus/sidebars.js

Rachel-Reverie committed:
adds flowcharts; removes the draft versions of the overview guide.

joshzzheng commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,148 @@
+
+import CheckpointFlowchart from './images_overview/checkpoint_flowchart.png'
+
+This guide is intended for new users of Great Expectations (GX) and those looking for a more foundational understanding of GX, its components, and its principal workflows.  You will be introduced to the moving parts of GX and how they interact without digging through the code that governs those processes and interactions.  The insight provided in this guide will help you understand the how and wh ...
```
nice, i like this

kwcanuck replied:
```suggestion
The information provided here is intended for new users of Great Expectations (GX) and those looking for an understanding of its components and its primary workflows. This overview of GX doesn’t require an in-depth understanding of the code that governs GX processes and interactions. This is an ideal place to start before moving to more advanced topics, or you want a better understanding of GX functionality.
```

Rachel-Reverie replied:
I've incorporated the majority of Rob's suggested edit.

joshzzheng commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,148 @@
+
+GX is a Python library that provides a framework for describing the acceptable state of data and then validating that the data meets those criteria. 
+
+GX enables you to build trust in your data's quality by testing how it performs against your expectations.  Reusable components and ongoing testing provide visibility into the health of your data pipeline over time. The human-readable reports generated by GX allow you to share the results of that analysis with others and better collaborate on improvements to both your data and your understanding  ...
```
what's the purpose of this paragraph? is it meant lay out the structure for the rest of the doc? if so, they should be in bullet points. overall feels a bit unnecessary to me.

joshzzheng commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,148 @@
+GX enables you to build trust in your data's quality by testing how it performs against your expectations.  Reusable components and ongoing testing provide visibility into the health of your data pipeline over time. The human-readable reports generated by GX allow you to share the results of that analysis with others and better collaborate on improvements to both your data and your understanding  ...
+
+With GX providing insight into your data and helping assure its quality, you can ensure that your data is fully suitable to be used as intended in your processes.
+
```
what's the purpose of the two paragraphs above? is it meant lay out the structure for the rest of the doc? if so, might be better if they're in bullet points. overall feels a bit unnecessary to me though.

Rachel-Reverie replied:
I was told a couple of times that we needed some kind of value-of-use statement at the start of the doc.  I'll remove this.

joshzzheng commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,148 @@
+When working with GX from end to end - that is, from installation of GX to the validation of your data - there are four core components that you will use.  These are "top level" components that interface, contain, or manage underlying objects and processes to make your work easier.  This guide will go over each of these four components and what they govern in more depth after this section. Howeve ...
+
+- **Data Context:** Manages the settings and metadata for a GX project, as well as providing an entry point to the GX Python API.
+- **Datasources:** tell GX how to connect to your source data, and how to organize the retrieved data for future use.
```
tells

joshzzheng commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,148 @@
+
+- [Data Context tutorials](/docs/guides/setup/configure_data_contexts_lp)
+
+### The GX API
```
We already mentioned you can interact with GX API via datacontext. I don't think we need another paragraph explaining all the stuff the GX public API can do - one sentence with a link to the api reference should suffice.

Rachel-Reverie replied:
I cut this down to the first two sentences.  I couldn't come up with a one-sentence version that I thought flowed well.

joshzzheng commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,148 @@
+
+### Settings and Stores
+
+Stores contain the metadata GX uses.  This includes things such as configurations for GX objects, information that is recorded when GX validates data, and credentials used for accessing data sources or remote environments.  GX utilizes one Store for each type of metadata, and the Data Context contains the settings that tell GX where that Store should reside and how to access it.
```
What are all the stores?

joshzzheng replied:
are settings and stores two different concepts? should we just remove talking about settings in general since that's kinda self explanatory?

Rachel-Reverie replied:
Settings and stores are different -- the settings are part of the set up process, and are all found in the same file.  The stores are individual files that are put to use throughout the rest of the GX processes.  I went ahead and removed the part about settings: I think I was holding it over from the more process-centric version of the guide.

joshzzheng commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,148 @@
+
+Stores contain the metadata GX uses.  This includes things such as configurations for GX objects, information that is recorded when GX validates data, and credentials used for accessing data sources or remote environments.  GX utilizes one Store for each type of metadata, and the Data Context contains the settings that tell GX where that Store should reside and how to access it.
+
+A Data Context includes default configurations for your Store locations. The default location of Stores in a Filesystem or Cloud Data Context is nested within the same folder hierarchy that contains the Data Context itself. An Ephemeral Data Context will default to keeping your Stores in memory. You can update these defaults to specify other storage locations, such as hosting them in a shared clo ...
```
I was pretty confused by this paragraph. We probably don't see to talk about folder locations and structures here.

Rachel-Reverie replied:
I removed the entire paragraph on settings.

joshzzheng commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,148 @@
+
+### Data Docs
+
+Data Docs are human-readable documentation generated by GX.  In general, Data Docs describe the standards that you expect your data to conform to, and the results of validating your data against those standards.  The storage and retrieval of this information is managed by the Data Context.
```
Data Docs are human-readable HTML pages generated by GX.
- Remove "In general"

joshzzheng commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,148 @@
+
+Data Docs are human-readable documentation generated by GX.  In general, Data Docs describe the standards that you expect your data to conform to, and the results of validating your data against those standards.  The storage and retrieval of this information is managed by the Data Context.
+
+Data Docs exist as webpages.  By default, a new Data Context includes a configuration for building them locally for File Data Contexts, in memory for Ephemeral Data Contexts, and online for Cloud Data Contexts.
```
- Remove "Data Docs exist as webpages."
- Can we rephrase the second sentence with Data Docs as the subject. I was confused by we're back to talking about Data Context again. Or maybe for configuration of Data Docs see this link should suffice.

Rachel-Reverie replied:
I removed the second sentence entirely.

joshzzheng commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,148 @@
+
+- [Stores tutorials](/docs/guides/setup/setup_overview_lp)
+
+### Data Docs
```
can we maybe use #### for these to make it more visually obviously that these concepts belong to Data Context?

Rachel-Reverie replied:
I don't really like jumping heading levels like that.  It would probably be better to update our CSS so that the change is throughout the docs.  I think that would be a separate PR, though, since the change would be docs-wide.

joshzzheng replied:
ok agreed

joshzzheng commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,148 @@
+
+When working with GX from end to end - that is, from installation of GX to the validation of your data - there are four core components that you will use.  These are "top level" components that interface, contain, or manage underlying objects and processes to make your work easier.  This guide will go over each of these four components and what they govern in more depth after this section. Howeve ...
+
+- **Data Context:** Manages the settings and metadata for a GX project, as well as providing an entry point to the GX Python API.
```
these should be links to the section headers

joshzzheng commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,148 @@
+
+## Datasources
+
+![Datasource overview](./images_overview/datasource_flowchart.png)
```
for this image, I thought we were going to use the Trino representation with the data connectors?

joshzzheng commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,148 @@
+
+Data Assets are collections of records within a Datasource.  While a Datasource tells GX how to connect to your source data, Data Assets tell GX how to organize that data.  For instance, you could define a Data Asset for a SQL Datasource as all the records within a specific table.  For a File Datasource, you could define a Data Asset as all of the `.csv` files in a specific subfolder.
+
+Data Assets allow you to organize your data into collections of records that do not necessarily conform to the schema of your source data.  For instance, in a SQL Datasource you can use a Query Data Asset to combine an arbitrary set of records into a single Data Asset.  Likewise, in a File Datasource you can use regular expressions (regex) to group a specific set of files into a single Data Asset ...
```
If possible, I'd like to combine this paragraph and the next 2 into one. 1 paragraph for data asset and 1 paragraph for batches.

Rachel-Reverie replied:
Done.

joshzzheng commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,148 @@
+
+Data Assets can be further partitioned into Batches.  Batches are unique subsets of records within a Data Asset.  For example, say you have a Data Asset in a SQL Datasource that consisted of all records from last year in a given table.  You could then partition those records into Batches of data that correspond to the records for individual months of that year.
+
+Using Batches to further partition a Data Asset is optional.  However, having a list for Batches in your Data Asset is not.  This is because GX uses Batches to specify the data to work with.  By default, if you do not further partition the data in a Data Asset it will contain a list with only one Batch.  In that case, the single Batch will consist of all the records in that Data Asset.
```
i dont think we need to introduce this idea here. i know it's really important but we can make that clear in a tutorial when actually working with assets and batches. i don't imagine someone reading this guide will remember this particularity when the time comes to implement anyways.

remember the purpose of conceptual guides is for understanding of concepts, not implementation.

Rachel-Reverie replied:
Removed.

joshzzheng commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,148 @@
+
+### Batch Requests
+
+A Batch Request specifies one or more Batches within the Data Asset.  Batch Requests are the primary way of retrieving data for use in GX.  Because Batch Requests can specify multiple Batches they provide significant flexibility in how you utilize the data in a single Data Asset.
```
My initial reaction is I thought Data Assets specify one or more batches.

I think we need to open with a sentence something like "Batch Requests retrieve Batches from Data Assets for validation".

joshzzheng commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,148 @@
+
+A Batch Request specifies one or more Batches within the Data Asset.  Batch Requests are the primary way of retrieving data for use in GX.  Because Batch Requests can specify multiple Batches they provide significant flexibility in how you utilize the data in a single Data Asset.
+
+As an example, GX can automate the process of running statistical analyzes for multiple Batches of data.  This is possible because you can provide a Batch Request that corresponds to multiple Batches in a Data Asset.  Alternatively, you can specify a single Batch from that same Data Asset so that you do not need to re-run the analyzes on all of your data when you are only interested in a single s ...
```
I'm a bit confused by the next two paragraphs. I think you're trying to say because we can retrieve multiple batches within a single batch request, we can run more sophisticated validations, such as expectations on some data across a period of time?

Rachel-Reverie replied:
Yes, that's exactly right.  It's used in Data Assistants to provide better Expectation suggestions.

joshzzheng commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,148 @@
+
+## Expectations
+
+An Expectation is a verifiable assertion about Source Data.  Similar to assertions in traditional Python unit tests, Expectations provide a flexible, declarative language for describing expected behaviors. Unlike traditional unit tests which describe the expected behavior of code given a specific input, Expectations apply to the input data itself. For example, you can define an Expectation that a ...
```
These two paragraphs read like general value prop for GX. I think keeping the first paragraph is fine or combine these two into one. We should have communicated this earlier in the funnel.

Rachel-Reverie replied:
Second paragraph removed.

joshzzheng commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,148 @@
+
+Expectation Suites are collections of Expectations describing your data.  When GX validates data, an Expectation Suite helps streamline the process by running all of the contained Expectations against that data.  In almost all cases, when you create an Expectation you will be creating it inside of an Expectation Suite object.
+
+You can define multiple Expectation Suites for the same data to cover different use cases.  An example could be having one Expectation Suite for raw data, and a more strict Expectation Suite for that same data post-processing.  Because an Expectation Suite is decoupled from a specific source of data, you can apply the same Expectation Suite against multiple, disparate Datasources.  For instance,  ...
```
what do you mean by same data? Same Data Asset? same Batch?

joshzzheng replied:
I really like these two paragraphs. One to explain what it is and one talks about applications of it.

joshzzheng commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,148 @@
+
+- [Expectations and Expectation Suite tutorials](/docs/guides/expectations/create_manage_expectations_lp)
+
+### Data Assistants
```
Are data assistants still going to be part of Expectations?

joshzzheng commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,148 @@
+- [Tutorials for creating custom Expectations](/docs/guides/expectations/custom_expectations_lp)
+- [Tutorials for using custom Expectations](/docs/guides/expectations/creating_custom_expectations/how_to_use_custom_expectations)
+
+## Checkpoints
```
love these. short and sweet.

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,148 @@
+
+This guide is intended for new users of Great Expectations (GX) and those looking for a more foundational understanding of GX, its components, and its principal workflows.  You will be introduced to the moving parts of GX and how they interact without digging through the code that governs those processes and interactions.  The insight provided in this guide will help you understand the how and wh ...
+
+## Great Expectations (GX)
```
```suggestion
## What is GX
```
You've already defined the acronym. I would remove this heading and include it with the overview statement.

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,148 @@
+
+## GX core components
+
+When working with GX from end to end - that is, from installation of GX to the validation of your data - there are four core components that you will use.  These are "top level" components that interface, contain, or manage underlying objects and processes to make your work easier.  This guide will go over each of these four components and what they govern in more depth after this section. Howeve ...
```
```suggestion
When working with GX you use the following four core components to access, store, and manage underlying objects and processes:
```

Rachel-Reverie replied:
suggestion implemented; it'll be in the next commit.

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,148 @@
+
+When working with GX from end to end - that is, from installation of GX to the validation of your data - there are four core components that you will use.  These are "top level" components that interface, contain, or manage underlying objects and processes to make your work easier.  This guide will go over each of these four components and what they govern in more depth after this section. Howeve ...
+
+- **Data Context:** Manages the settings and metadata for a GX project, as well as providing an entry point to the GX Python API.
```
```suggestion
- **Data Context:** Manages the settings and metadata for a GX project, and provides an entry point to the GX Python API.
```

Rachel-Reverie replied:
suggestion implemented; it'll be in the next commit.

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,148 @@
+When working with GX from end to end - that is, from installation of GX to the validation of your data - there are four core components that you will use.  These are "top level" components that interface, contain, or manage underlying objects and processes to make your work easier.  This guide will go over each of these four components and what they govern in more depth after this section. Howeve ...
+
+- **Data Context:** Manages the settings and metadata for a GX project, as well as providing an entry point to the GX Python API.
+- **Datasources:** tell GX how to connect to your source data, and how to organize the retrieved data for future use.
```
```suggestion
- **Datasources:** Connects GX to your source data, and organizes retrieved data for future use.
```

Rachel-Reverie replied:
suggestion implemented; it'll be in the next commit.

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,148 @@
+
+- **Data Context:** Manages the settings and metadata for a GX project, as well as providing an entry point to the GX Python API.
+- **Datasources:** tell GX how to connect to your source data, and how to organize the retrieved data for future use.
+- **Expectations:** tell GX the standards to which your data should conform.
```
```suggestion
- **Expectations:** Identifies the standards to which your data should conform.
```

Rachel-Reverie replied:
suggestion implemented; it'll be in the next commit.

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,148 @@
+- **Data Context:** Manages the settings and metadata for a GX project, as well as providing an entry point to the GX Python API.
+- **Datasources:** tell GX how to connect to your source data, and how to organize the retrieved data for future use.
+- **Expectations:** tell GX the standards to which your data should conform.
+- **Checkpoints:** validate a set of Expectations against a specific set of data.
```
```suggestion
- **Checkpoints:** Validates a set of Expectations against a specific set of data.
```
Explanations should be parallel.

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,148 @@
+- **Datasources:** tell GX how to connect to your source data, and how to organize the retrieved data for future use.
+- **Expectations:** tell GX the standards to which your data should conform.
+- **Checkpoints:** validate a set of Expectations against a specific set of data.
+
```
```suggestion
```

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,148 @@
+
+![A Data Context](./images_overview/data_context_flowchart.png)
+
+A Data Context defines a GX project.  In Python, the Data Context object serves as your entry point for the GX API and manages various classes to limit the objects you need to directly manage yourself.  A Data Context contains all of the metadata used by GX, the configurations for GX objects, and the output from validating data.
```
```suggestion
A Data Context manages the settings and metadata for a GX project. In Python, the Data Context object serves as an entry point for the GX API and manages various classes to limit the objects you need to directly manage.  A Data Context contains all of the metadata used by GX, the configurations for GX objects, and the output from validating data.
```

Rachel-Reverie replied:
suggestion implemented; it'll be in the next commit.

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,148 @@
+
+### The GX API
+
+A Data Context object in Python provides convenience methods to further configure and interact with GX.  These methods and the objects and additional methods accessed through them compose the GX public API.  
```
```suggestion
A Data Context object in Python provides a convenient method for configuring and interacting with GX.  These methods and the objects and additional methods accessed through them compose the GX public API.  
```

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,148 @@
+
+### Settings and Stores
+
+Stores contain the metadata GX uses.  This includes things such as configurations for GX objects, information that is recorded when GX validates data, and credentials used for accessing data sources or remote environments.  GX utilizes one Store for each type of metadata, and the Data Context contains the settings that tell GX where that Store should reside and how to access it.
```
```suggestion
Stores contain the metadata GX uses.  This includes configurations for GX objects, information that is recorded when GX validates data, and credentials used for accessing data sources or remote environments.  GX utilizes one Store for each type of metadata, and the Data Context contains the settings that tell GX where that Store should reside and how to access it.
```

Rachel-Reverie replied:
suggestion implemented; it'll be in the next commit.

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,148 @@
+
+### Data Docs
+
+Data Docs are human-readable documentation generated by GX.  In general, Data Docs describe the standards that you expect your data to conform to, and the results of validating your data against those standards.  The storage and retrieval of this information is managed by the Data Context.
```
```suggestion
Data Docs are human-readable documentation generated by GX.  Data Docs describe the standards that you expect your data to conform to, and the results of validating your data against those standards. The Data Context manages the storage and retrieval of this information.
```

Rachel-Reverie replied:
suggestion implemented; it'll be in the next commit.

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,148 @@
+
+Data Docs are human-readable documentation generated by GX.  In general, Data Docs describe the standards that you expect your data to conform to, and the results of validating your data against those standards.  The storage and retrieval of this information is managed by the Data Context.
+
+Data Docs exist as webpages.  By default, a new Data Context includes a configuration for building them locally for File Data Contexts, in memory for Ephemeral Data Contexts, and online for Cloud Data Contexts.
```
```suggestion
Data Docs are presented as webpages.  By default, a new Data Context includes a configuration for building them locally for File Data Contexts, in memory for Ephemeral Data Contexts, and online for Cloud Data Contexts.
```

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,148 @@
+
+![Datasource overview](./images_overview/datasource_flowchart.png)
+
+Datasources connect GX to your source data.  Your source data could be CSV files in a folder, a PostgreSQL database hosted on AWS, or any combination of numerous data formats and environments.  Regardless of your source data's format and where it resides, Datasources provide GX with a unified API for working with it.
```
```suggestion
Datasources connect GX to source data such as CSV files in a folder, a PostgreSQL database hosted on AWS, or any combination of data formats and environments. Regardless of the format of your source data or where it resides, Datasources provide GX with a unified API for working with it.
```

Rachel-Reverie replied:
suggestion implemented; it'll be in the next commit.

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,148 @@
+
+### Actions
+
+One of the most powerful features of Checkpoints is that you can configure them to run Actions. The Validation Results generated when a Checkpoint runs determine what Actions are performed. Typical use cases include sending email, Slack, or custom notifications. Another common use case is updating Data Docs sites. Actions can be used to do anything you are capable of programming in Python. Action ...
```
```suggestion
One of the most powerful features of Checkpoints is that you can configure them to run Actions. The Validation Results generated when a Checkpoint runs determine what Actions are performed. Typical use cases include sending email, Slack messages, or custom notifications. Another common use case is updating Data Docs sites. Actions can be used to do anything you are capable of programming in Python. Actions are a versatile tool for integrating Checkpoints in your pipeline's workflow.
```

Rachel-Reverie replied:
suggestion implemented; it'll be in the next commit.

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,148 @@
+
+- [Checkpoint tutorials](/docs/guides/validation/checkpoints/checkpoint_lp)
+
+### Validation Results
```
```suggestion
### Validation results
```

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,148 @@
+
+![Checkpoint overview](./images_overview/checkpoint_flowchart.png)
+
+A Checkpoint is the primary means for validating data in a production deployment of GX. Checkpoints provide an abstraction for bundling a Batch (or Batches) of data with an Expectation Suite (or several), and then running those Expectations against the paired data.
```
```suggestion
A Checkpoint is the primary means for validating data in a production deployment of GX. Checkpoints provide an abstraction for bundling a Batch (or Batches) of data with one or more Expectation Suites, and then running those Expectations against the paired data.
```

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,148 @@
+
+### Custom Expectations
+
+GX comes with a built-in library of more than 50 common expectations.  However, you can also create custom Expectations if you need something more specialized for your data validation.  You can even use custom Expectations that have been contributed by other GX community members by installing them as Plugins.
```
```suggestion
 A built-in library of more than 50 common expectations are included with GX.  However, you can create custom Expectations if you need a more specialized for data validation.  You can also use custom Expectations that have been contributed by other GX community members by installing them as Plugins.
```

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,148 @@
+
+### Data Assistants
+
+A Data Assistant is a utility that automates the process of building Expectations by asking questions about your data, gathering information to describe what is observed, and then presenting Metrics and proposed Expectations based on the answers it has determined.  This can greatly accelerate the process of creating Expectations for the provided data.
```
```suggestion
A Data Assistant is a utility that automates the process of building Expectations by asking questions about your data, gathering information to describe what is observed, and then presenting Metrics and proposed Expectations based on the answers it has determined.  This can accelerate the process of creating Expectations for the provided data.
```

Rachel-Reverie replied:
suggestion implemented; it'll be in the next commit.

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,148 @@
+- **Ephemeral Data Context:** Exists in memory, and does not persist beyond the current Python session.
+- **File Data Context:** Exists as a folder and configuration files. Its contents persist between Python sessions.
+- **Cloud Data Context:** Supports persistence between Python sessions, but additionally serves as the entry point for Great Expectations Cloud.
+
```
```suggestion
```

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,148 @@
+- **File Data Context:** Exists as a folder and configuration files. Its contents persist between Python sessions.
+- **Cloud Data Context:** Supports persistence between Python sessions, but additionally serves as the entry point for Great Expectations Cloud.
+
+- [Data Context tutorials](/docs/guides/setup/configure_data_contexts_lp)
```
The linked topic isn't a tutorial.

I would remove this as bullet point from the list. Instead, introduce the information. For example:

To instantiate and convert a Data Context, see [Configure Data Contexts](/docs/guides/setup/configure_data_contexts_lp).

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,148 @@
+
+Through the GX public API you will be able to connect GX to your data sources, define the standards to which you expect your data to conform, retrieve data from your data sources, and validate that data against your defined standards.  When working with GX in Python, you will almost always start by instantiating an existing or new Data Context with which to access the GX public API.
+
+- [The GX API reference](/docs/reference/api_reference)
```
```suggestion
For more information about the GX API, see the [GX API reference](/docs/reference/api_reference).
```

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,148 @@
+
+A Data Context includes default configurations for your Store locations. The default location of Stores in a Filesystem or Cloud Data Context is nested within the same folder hierarchy that contains the Data Context itself. An Ephemeral Data Context will default to keeping your Stores in memory. You can update these defaults to specify other storage locations, such as hosting them in a shared clo ...
+
+- [Stores tutorials](/docs/guides/setup/setup_overview_lp)
```
```suggestion
To configure a Store, see [Configure your Great Expectations environment](/docs/guides/setup/setup_overview_lp).
```

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,148 @@
+
+You can configure where your Data Docs are hosted.  Unlike Stores, you can define configurations for multiple Data Docs sites.  You can also specify what information each Data Doc site provides, allowing you to format and provide different Data Docs for different use cases.
+
+- [Data Docs tutorial](/docs/guides/setup/configuring_data_docs/host_and_share_data_docs)
```
```suggestion
To host and share Data Docs, see [Host and share Data Docs](/docs/guides/setup/configuring_data_docs/host_and_share_data_docs)
```
Again, you're not linking to a tutorial. You shouldn't be referring to the topics as tutorials. This information shouldn't appear as a single item in a bulleted list.

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,148 @@
+
+Using Batches to further partition a Data Asset is optional.  However, having a list for Batches in your Data Asset is not.  This is because GX uses Batches to specify the data to work with.  By default, if you do not further partition the data in a Data Asset it will contain a list with only one Batch.  In that case, the single Batch will consist of all the records in that Data Asset.
+
+- [Data Asset and Batch tutorials](/docs/guides/connecting_to_your_data/manage_data_assets_lp)
```
```suggestion
For more information about connecting to source data, see [Connect to Source Data](/docs/guides/connecting_to_your_data/manage_data_assets_lp).
```

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,148 @@
+
+Taking the example of a Data Asset that has been partitioned into Batches by months, this would let you build a statistical model off of each month of your existing data by providing all of the Batches in a single Batch Request.  Then, after an update, you could specify that you only wanted to run your analysis on the most recent month's data by providing a Batch Request that only indicates that  ...
+
+- [Batch Request tutorial](/docs/guides/connecting_to_your_data/fluent/batch_requests/how_to_request_data_from_a_data_asset)
```
```suggestion
For more information about Batch Requests, see  [Manage Data Assets](/docs/guides/connecting_to_your_data/fluent/batch_requests/how_to_request_data_from_a_data_asset).
```

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,148 @@
+Expectations can be built directly from the domain knowledge of subject matter experts, interactively while introspecting a set of data, or through automated tools provided by GX.
+
+For a list of available Expectations, see [the Expectation Gallery](https://greatexpectations.io/expectations/).
+
```
```suggestion
```

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,148 @@
+For a list of available Expectations, see [the Expectation Gallery](https://greatexpectations.io/expectations/).
+
+
+- [Expectation Tutorials](/docs/guides/expectations/expectations_lp)
```
```suggestion
To create and manage Expectations and Expectation Suites, see [Create Expectations](/docs/guides/expectations/expectations_lp).
```

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,148 @@
+
+A Data Assistant is a utility that automates the process of building Expectations by asking questions about your data, gathering information to describe what is observed, and then presenting Metrics and proposed Expectations based on the answers it has determined.  This can greatly accelerate the process of creating Expectations for the provided data.
+
+- [Data Assistant tutorials](/docs/guides/expectations/profilers_data_assistants_lp)
```
```suggestion
To Cceate an Expectation Suite with the Onboarding Data Assistant, see [Profilers and Data Assistants](/docs/guides/expectations/profilers_data_assistants_lp).
```

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,148 @@
+
+GX comes with a built-in library of more than 50 common expectations.  However, you can also create custom Expectations if you need something more specialized for your data validation.  You can even use custom Expectations that have been contributed by other GX community members by installing them as Plugins.
+
+- [Tutorials for creating custom Expectations](/docs/guides/expectations/custom_expectations_lp)
```
```suggestion
To create and manage Custom Expectations, see [Create and manage Custom Expectations](/docs/guides/expectations/custom_expectations_lp).  To use a Custom Expectation, see [Use a Custom Expectation](/docs/guides/expectations/creating_custom_expectations/how_to_use_custom_expectations).
```

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,148 @@
+GX comes with a built-in library of more than 50 common expectations.  However, you can also create custom Expectations if you need something more specialized for your data validation.  You can even use custom Expectations that have been contributed by other GX community members by installing them as Plugins.
+
+- [Tutorials for creating custom Expectations](/docs/guides/expectations/custom_expectations_lp)
+- [Tutorials for using custom Expectations](/docs/guides/expectations/creating_custom_expectations/how_to_use_custom_expectations)
```
```suggestion
```

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,148 @@
+
+A Checkpoint is the primary means for validating data in a production deployment of GX. Checkpoints provide an abstraction for bundling a Batch (or Batches) of data with an Expectation Suite (or several), and then running those Expectations against the paired data.
+
+- [Checkpoint tutorials](/docs/guides/validation/checkpoints/checkpoint_lp)
```
```suggestion
To manage your Checkpoints, see [Manage Checkpoints](/docs/guides/validation/checkpoints/checkpoint_lp).
```

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,148 @@
+
+The Validation Results returned by GX tell you how your data corresponds to what you expected of it. You can view this information in the Data Docs that are configured in your Data Context. Evaluating your Validation Results helps you identify issues with your data. If the Validation Results show that your data meets your Expectations, you can confidently use it.
+
+- [Validation Result further reading](/docs/terms/validation_result)
```
```suggestion
To learn more about Validation Results, see [Validation Result](/docs/terms/validation_result).
```

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,148 @@
+
+One of the most powerful features of Checkpoints is that you can configure them to run Actions. The Validation Results generated when a Checkpoint runs determine what Actions are performed. Typical use cases include sending email, Slack, or custom notifications. Another common use case is updating Data Docs sites. Actions can be used to do anything you are capable of programming in Python. Action ...
+
+- [Actions tutorials](docs/guides/validation/validation_actions/actions_lp)
```
```suggestion
To learn more about using Actions to send Validation Result notifications, update Data Docs, and store Validation Results, see [Configure Actions](docs/guides/validation/validation_actions/actions_lp/).
```

kwcanuck commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,148 @@
+
+Data Assets allow you to organize your data into collections of records that do not necessarily conform to the schema of your source data.  For instance, in a SQL Datasource you can use a Query Data Asset to combine an arbitrary set of records into a single Data Asset.  Likewise, in a File Datasource you can use regular expressions (regex) to group a specific set of files into a single Data Asset ...
+
+Data Assets can be further partitioned into Batches.  Batches are unique subsets of records within a Data Asset.  For example, say you have a Data Asset in a SQL Datasource that consisted of all records from last year in a given table.  You could then partition those records into Batches of data that correspond to the records for individual months of that year.
```
```suggestion
Data Assets can be further partitioned into Batches.  Batches are unique subsets of records within a Data Asset.  For example, say you have a Data Asset in a SQL Datasource that consists of all records from last year in a given table.  You could then partition those records into Batches of data that correspond to the records for individual months of that year.
```

Rachel-Reverie replied:
suggestion implemented; it'll be in the next commit.

Rachel-Reverie committed:
- updates per feedback

Rachel-Reverie committed:
Merge branch 'develop' into d/doc_483/consolidated_gx_overview_conceptual_guide

r34ctor commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,132 @@
+
+### Data Assets and Batches
+
+Data Assets are collections of records within a Datasource.  While a Datasource tells GX how to connect to your source data, Data Assets tell GX how to organize that data. Although the records in your Data Assets can correspond directly to the contents of tables or files in your source data they do not necessarily need to. For instance, you could combine multiple tables worth of records in a SQL  ...
```
I think it would be helpful to have a more concrete example cited here and perhaps why you may want to do it.

r34ctor commented on docs/docusaurus/docs/conceptual_guides/gx_overview.md:
```diff
@@ -0,0 +1,132 @@
+
+A Batch Request specifies one or more Batches within the Data Asset.  Batch Requests are the primary way of retrieving data for use in GX.  Because Batch Requests can retrieve multiple Batches they provide significant flexibility in how you utilize the data in a single Data Asset.
+
+As an example, GX can automate the process of running statistical analyzes for multiple Batches of data.  This is possible because you can provide a Batch Request that corresponds to multiple Batches in a Data Asset.  Alternatively, you can specify a single Batch from that same Data Asset so that you do not need to re-run the analyzes on all of your data when you are only interested in a single s ...
```
```suggestion
As an example, GX can automate the process of running statistical analyses for multiple Batches of data.  This is possible because you can provide a Batch Request that corresponds to multiple Batches in a Data Asset.  Alternatively, you can specify a single Batch from that same Data Asset so that you do not need to re-run the analyses on all of your data when you are only interested in a single subset.
```

r34ctor reviewed (commented):
From a newbie's perspective, this is a really nice overview of all of the components and workflows. I would say this is great for a v1. One suggestion I have that could make the experience better though is to perhaps include a simple usecase that you can refer back to within each section, so that the reader can fit the pieces of the puzzle in their head a little bit better. This would be particularly helpful when talking about the Datasources, Data Assets, Batches, etc.

Rachel-Reverie committed:
Merge branch 'develop' into d/doc_483/consolidated_gx_overview_conceptual_guide

Rachel-Reverie committed:
- updates per feedback

Rachel-Reverie committed:
Merge remote-tracking branch 'origin/d/doc_483/consolidated_gx_overview_conceptual_guide' into d/doc_483/consolidated_gx_overview_conceptual_guide

Rachel-Reverie committed:
Merge branch 'develop' into d/doc_483/consolidated_gx_overview_conceptual_guide

github-actions[bot] committed:
Merge develop into d/doc_483/consolidated_gx_overview_conceptual_guide
